torch/distributed/distributed_c10d.py

```diff
@@ -2139,17 +2169,27 @@ def all_gather_coalesced(
     ]
 
     if group is None:
-        default_pg = _get_default_group()
-        work = default_pg.allgather_coalesced(output_tensor_lists, input_tensor_list)
-    else:
-        work = group.allgather_coalesced(output_tensor_lists, input_tensor_list)
-
+        group = _get_default_group()
+
+    flatten_tensors = [x.flatten() for x in input_tensor_list]
+    coalesced_input = torch.cat(flatten_tensors)
+    coalesced_output = [torch.zeros(coalesced_input.shape) for _ in range(len(output_tensor_lists))]
```

Comment:
I believe we also need to respect the device type here. So, I think we should use `torch.empty_like(coalesced_input)` instead